zerver/webhooks/newrelic/view.py

```diff
@@ -1,169 +1,278 @@
 
 from zerver.decorator import webhook_view
-from zerver.lib.exceptions import JsonableError
+from zerver.lib.exceptions import UnsupportedWebhookEventTypeError
 from zerver.lib.response import json_success
 from zerver.lib.typed_endpoint import JsonBodyPayload, typed_endpoint
 from zerver.lib.validator import (
     WildValue,
+    check_float,
     check_int,
     check_list,
     check_none_or,
     check_string,
-    check_string_fixed_length,
+    check_string_in,
     check_union,
 )
 from zerver.lib.webhooks.common import check_send_webhook_message, unix_milliseconds_to_timestamp
 from zerver.models import UserProfile
 
-# Newrelic planned to upgrade Alert Notification Channels to Workflows and Destinations
-# https://discuss.newrelic.com/t/plan-to-upgrade-alert-notification-channels-to-workflows-and-destinations/188205
-# This view will handle both old and new format but will keep it easy to delete the old code
-# once it is EOLed by the end of June, 2023
+PRIORITIES = {
+    "CRITICAL": ":red_circle:",
+    "HIGH": ":orange_circle:",
+    "MEDIUM": ":yellow:",
+    "LOW": ":blue_circle:",
+}
+
+TIME_STAMPS_MAP: Dict[str, str] = {
+    "CREATED": "createdAt",
+    "ACTIVATED": "activatedAt",
+    "UPDATED": "updatedAt",
+    "CLOSED": "closedAt",
+    "ACKNOWLEDGED": "acknowledgedAt",
+}
+
+TIME_STAMP_STR: Dict[str, Optional[str]] = {}
+
+INCIDENT_STATES = ["CREATED", "ACTIVATED", "CLOSED"]
+
+DEFAULT_NEWRELIC_URL = "https://one.newrelic.com/alerts-ai"
+
+INTEGRATION_LAST_UPDATED = "April 2024"
 
-# Once old is EOLed, delete the OPEN_TEMPLATE
-OPEN_TEMPLATE = """
-[Incident]({incident_url}) **opened** for condition: **{condition_name}** at <time:{iso_timestamp}>
-``` quote
-{details}
+BASE_NOTIFICATION_WRAP_TEMPLATE = """
+```spoiler {header}
+
+{body}
 ```
-""".strip()
+"""
+
+NOTIFICATION_HEADER = "{priority_symbol} {priority} **priority [issue]({incident_url}) has been **{status}** at** <time: {time_updated_str} >"
+
+NOTIFICATION_BODY_TEMPLATE = """
+| :file:  **Incident details**  |
+|:--------|
+|**[{title}]({incident_url})**   |
+|:checkbox:  Alert Policy: {alert_policy_str}|
+|:spiral_notepad: Conditions: {conditions_str}|
+|:warning: Total Incidents: **{total_incidents}**|
+|:clock: Incident Created At: <time: {time_created_str} > |
+"""
 
-ACTIVE_TEMPLATE = """
-[Incident]({incident_url}) **active** for condition: **{condition_name}** at <time:{iso_timestamp}>
-``` quote
-{details}
+TOPIC_TEMPLATE = "{state} {priority} priority"
+
+FALL_BACK_NOTIFICATION_TEMPLATE = """
+```spoiler :danger: An [Incident](https://one.newrelic.com/alerts-ai) updated
+
+**Warning: Unable to send the default notification format**
+At least one essential field is missing from the payload.
+Try to reset the payload template to the default from
+New Relic if you're unsure.
+Please reach out to the [the Zulip development community](https://chat.zulip.org/#narrow/stream/127-integrations)
+if you think this integration is out-of-date.
```

Comment:
Yeah, I apply the line breaks as such to make the overall structure of the fallback notification closely resemble the other notification messages by making them roughly as wide as the other notification message. The bit that seemed off to me was that there isn't an empty line between line 76 and 77. So adjusting that, this is closer to what I intend by structuring the line breaks as such:
![image](https://github.com/zulip/zulip/assets/108744694/14dc8ef0-3e6c-4f8d-9bc3-2bcb12b4337a)

And I don't think it's _horrible_ without line breaks as one paragraph. Here's how it would compare:
![image](https://github.com/zulip/zulip/assets/108744694/47ff39d5-7a4d-4788-9cc0-c13f84a0d732)

This notification format hasn't gotten input from the [CZO](https://chat.zulip.org/#narrow/stream/101-design/topic/.2329730.20Updating.20Newrelic.20Notification) discussion about the notification design, so opinions are appreciated!